### PR TITLE
Allow plugins to disable background generation

### PR DESCRIPTION
**Pull request by Frago9876543210**
## Introduction
See title

### API changes
```diff
- Server::generateLevel(string $name, int $seed = null, $generator = null, array $options = [])
+ Server::generateLevel(string $name, int $seed = null, $generator = null, array $options = [], bool $backgroundGeneration = true)
```

### CODE REVIEW

File: src/pocketmine/Server.php [hunk 1]
```diff
@@ -1048,10 +1048,11 @@ public function loadLevel(string $name) : bool{
 	 * @param int|null    $seed
 	 * @param string|null $generator Class name that extends pocketmine\level\generator\Generator
 	 * @param array       $options
+	 * @param bool        $backgroundGeneration
 	 *
 	 * @return bool
 	 */
-	public function generateLevel(string $name, int $seed = null, $generator = null, array $options = []) : bool{
+	public function generateLevel(string $name, int $seed = null, $generator = null, array $options = [], $backgroundGeneration = true) : bool{
```
Comment (dktapps) [inline src/pocketmine/Server.php]:
```suggestion
	public function generateLevel(string $name, int $seed = null, $generator = null, array $options = [], bool $backgroundGeneration = true) : bool{
```

**Review by dktapps — CHANGES_REQUESTED** [review]
Looks fine after nits are fixed.

**Review by dktapps — APPROVED** [review]

**Comment by dktapps** [comment]
This might have unintended side effects on world loading. If no spawn chunks are generated, the initial region on Region-based worlds will not be created, which will prevent the format selector detecting the correct world format. Plugins would have to set chunks into the world before unloading it, or risk the empty world getting overwritten with a new one.

**Comment by Frago9876543210** [comment]
Yes, it has side effects, but the responsibility lies with the plugins. This can simplify the format conversion, which is required later.

**Pull request merged**